File: instalacao-ci/application/models/Entity/Votacao.java
package br.edu.unipampa.app.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Objects;

@Entity(name = "votacoes")
public class Votacao implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private boolean votacaoAberta = false;

    @OneToOne(cascade = CascadeType.ALL, orphanRemoval = true)
    private Urna urna;

    @OneToOne(cascade = CascadeType.REFRESH)
    @JsonIgnore
    private ItemDePauta itemDePauta;

    @OneToOne(cascade = CascadeType.ALL)
    private ResultadoDaVotacao resultado;

    @Enumerated
    private StatusVotacao status = StatusVotacao.FECHADA;

    @Enumerated
    private Turno turno;

    public Votacao() {

    }

    public Votacao(Urna urna, ItemDePauta itemDePauta) {
        this.setUrna(urna);
        this.setItemDePauta(itemDePauta);
    }

    public void registraVotoNaUrna(Voto voto) {
        this.getUrna().recebe(voto);
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public boolean isVotacaoAberta() {
        return votacaoAberta;
    }

    public void setVotacaoAberta(boolean votacaoAberta) {
        this.votacaoAberta = votacaoAberta;
    }

    public Urna getUrna() {
        return urna;
    }

    public void setUrna(Urna urna) {
        this.urna = urna;
    }

    public ItemDePauta getItemDePauta() {
        return itemDePauta;
    }

    public void setItemDePauta(ItemDePauta itemDePauta) {
        this.itemDePauta = itemDePauta;
    }

    public ResultadoDaVotacao getResultado() {
        return resultado;
    }

    public void setResultado(ResultadoDaVotacao resultado) {
        this.resultado = resultado;
    }

    public StatusVotacao getStatus() {
        return status;
    }

    public void setStatus(StatusVotacao status) {
        this.status = status;
    }

    public Turno getTurno() {
        return turno;
    }

    public void setTurno(Turno turno) {
        this.turno = turno;
    }

    @Override
    public String toString() {
        return "Votacao{" +
                "id=" + id +
                ", votacaoAberta=" + votacaoAberta +
                ", urna=" + urna +
                ", itemDePauta=" + itemDePauta +
                ", resultado=" + resultado +
                ", status=" + status +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Votacao votacao = (Votacao) o;
        return isVotacaoAberta() == votacao.isVotacaoAberta() &&
                Objects.equals(getId(), votacao.getId()) &&
                Objects.equals(getUrna(), votacao.getUrna()) &&
                Objects.equals(getItemDePauta(), votacao.getItemDePauta()) &&
                Objects.equals(getResultado(), votacao.getResultado()) &&
                getStatus() == votacao.getStatus();
    }

    @Override
    public int hashCode() {

        return Objects.hash(getId(), isVotacaoAberta(), getUrna(), getItemDePauta(), getResultado(), getStatus());
    }

}

File: instalacao-ci/application/models/Entity/Votacao.php
<?php

namespace Entity;


/**
 * @Entity
 * @Table(name="votacoes")
 */
class Votacao
{

    /**
     * @Id
     * @Column(type="integer", nullable=false)
     * @GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @Column(type="boolean", nullable=false)
     */
    private $votacaoAberta = false;

    /**
     * @OneToOne(targetEntity="Urna", mappedBy="urnas", cascade={"persist", "remove", "merge"}, orphanRemoval=true)
     * @JoinColumn(name="urna_id", referencedColumnName="id")
     */
    private $urna;

    /**
     * @OneToOne(targetEntity="ItemDePauta")
     * @JoinColumn(name="itemdepauta_id", referencedColumnName="id")
     */
    private $itemDePauta;

    /**
     * @OneToOne(targetEntity="ResultadoDaVotacao", cascade={"persist", "remove", "merge", "refresh"})
     */
    private $resultado;

    /**
     * @Column(type="string", nullable=false)
     */
    private $status = StatusVotacao::FECHADA;

    /**
     * @Column(type="string", nullable=false)
     */
    private $turno;

    /**
     * Votacao constructor.
     * @param $urna
     * @param $itemDePauta
     */
    public function __construct(Urna $urna, ItemDePauta $itemDePauta)
    {
        $this->urna = $urna;
        $this->itemDePauta = $itemDePauta;
    }

    /**
     * @return mixed
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * @param mixed $id
     */
    public function setId($id)
    {
        $this->id = $id;
    }

    /**
     * @return mixed
     */
    public function getVotacaoAberta()
    {
        return $this->votacaoAberta;
    }

    /**
     * @param mixed $votacaoAberta
     */
    public function setVotacaoAberta($votacaoAberta)
    {
        $this->votacaoAberta = $votacaoAberta;
    }

    /**
     * @return mixed
     */
    public function getUrna()
    {
        return $this->urna;
    }

    /**
     * @param mixed $urna
     */
    public function setUrna($urna)
    {
        $this->urna = $urna;
    }

    /**
     * @return mixed
     */
    public function getItemDePauta()
    {
        return $this->itemDePauta;
    }

    /**
     * @param mixed $itemDePauta
     */
    public function setItemDePauta($itemDePauta)
    {
        $this->itemDePauta = $itemDePauta;
    }

    /**
     * @return mixed
     */
    public function getResultado()
    {
        return $this->resultado;
    }

    /**
     * @param mixed $resultado
     */
    public function setResultado($resultado)
    {
        $this->resultado = $resultado;
    }

    /**
     * @return mixed
     */
    public function getStatus()
    {
        return $this->status;
    }

    /**
     * @param mixed $status
     */
    public function setStatus($status)
    {
        if (!in_array($status, StatusVotacao::getAvailableTypes())) {
            throw new \InvalidArgumentException("Invalid type");
        }

        $this->status = $status;
    }

    /**
     * @return mixed
     */
    public function getTurno()
    {
        return $this->turno;
    }

    /**
     * @param mixed $turno
     */
    public function setTurno($turno)
    {
        if (!in_array($turno, Turno::getAvailableTypes())) {
            throw new \InvalidArgumentException("Invalid type");
        }

        $this->turno = $turno;
    }

//    public void registraVotoNaUrna(Voto voto) {
//        this.getUrna().recebe(voto);
//    }

}
